keps/sig-node/3721-support-for-env-files/README.md

```diff
@@ -289,25 +289,50 @@ type FileKeySelector struct {
 
 The full specification of an env file:
 
-1. **File Format**: The environment variable (env) file must adhere to valid env file syntax to ensure proper parsing. The syntax for env files is as follows:
-   * Blank Lines: Blank lines are ignored.
-   * Leading Spaces: Leading spaces on all lines are ignored.
-   * Variable Declaration: Variables must be declared as `VAR=VAL`. Spaces surrounding `=` and trailing spaces are ignored.
-       ```
-       VAR=VAL → VAL
-       ```
-   * Comments: Lines beginning with # are treated as comments and ignored.
-       ```
-       # comment
-       VAR=VAL → VAL
-       VAR=VAL # not a comment → VAL # not a comment
-       ```
-   * Line Continuation: A backslash (`\`) at the end of a variable declaration line indicates the value continues on the next line. The lines are joined with a single space.
-       ```
-       VAR=VAL \
-       VAL2
-       → VAL VAL2
-       ```
+1. **Environment Variable Syntax**:
+
+    The syntax for environment variables is a strict subset of the POSIX shell standard, with one key constraint: variable values must be enclosed in single quotes.
+
+    For example:
+    ```
+    MY_VAR='my-literal-value'
+    ```
+
+    **Parsing Rationale and POSIX Compliance:**
+
+    When parsing the variable's value, our interpreter will strictly emulate the behavior of a double-quoted variable expansion ("$VAR") in a POSIX-compliant shell. This design choice ensures that the provided value is treated as a literal string without any interpretation or modification.
+
+    To illustrate the difference in a POSIX shell:
+
+    Given the following variable definition with a newline character:
+    ```
+    VAR='line1
+    line2'
+    ```
+    * An unquoted expansion (echo $VAR) subjects the value to word splitting. The shell replaces the newline with a space, resulting in a single line of output:
```

Comment:
I think we should not include this detail -- the variable value includes the newline, period.  How shells do variable expansion is entirely up to them.  If someone read this in Go with `os.Getenv()` they would see the newline.

Say less :)